### PR TITLE
[ci] Don't run tests on internal pipeline builds

### PR DESCRIPTION
**Pull request by radical**
## Description

Tests for `main` are run on the public pipeline, and are not needed for the internal one where we essentially build the packages to be published.

## Checklist

- Is this feature complete?
  - [ ] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [ ] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [ ] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [ ] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Link to aspire-docs issue: 
  - [ ] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6413)

### CODE REVIEW

**Comment by radical** [comment]
Internal build - https://dev.azure.com/dnceng/internal/_build/results?buildId=2565704&view=results .

**Comment by radical** [comment]
@joperezr this one is ready for review.

**Review by eerhardt — APPROVED** [review]
LGTM. Thanks!

File: eng/pipelines/templates/BuildAndTest.yml [hunk 1]
```diff
@@ -21,15 +21,14 @@ parameters:
...
-  # Internal pipeline: Build with pack+sign+archive
+  # Internal pipeline: Build with pack+sign
   - ${{ if ne(parameters.runAsPublic, 'true') }}:
     - script: ${{ parameters.buildScript }}
               -restore -build
               -pack
               -sign $(_SignArgs)
               -publish $(_PublishArgs)
               -configuration ${{ parameters.buildConfig }}
-              /p:ArchiveTests=true
```
Comment (joperezr) [inline eng/pipelines/templates/BuildAndTest.yml:32]: I'm not following, why are we no longer zipping the tests?

Comment (eerhardt) [reply eng/pipelines/templates/BuildAndTest.yml:32]: Why do we need to zip tests if we aren't giong to run them?

Comment (radical) [reply eng/pipelines/templates/BuildAndTest.yml:32]: Yep, the tests are zipped only for sending to helix.

Comment (joperezr) [reply eng/pipelines/templates/BuildAndTest.yml:32]: I thought this template was used by both our public CI and official builds, but maybe that's not the case then?

Comment (joperezr) [reply eng/pipelines/templates/BuildAndTest.yml:32]: Ohhhhh just saw the condition above for not run as public... duh! don't mind me, I can't read apparently 😝

Comment (radical) [reply eng/pipelines/templates/BuildAndTest.yml:32]: There is a conditional step below this for public builds.

**Review by joperezr — APPROVED** [review]

**Comment by joperezr** [comment]
Merging and backporting this to release/9.0 as it is needed for the internal builds too. Thanks @radical!

**Pull request merged**